### PR TITLE
Adds the ringbell and maps it into the boxing simulation

### PR DESCRIPTION
**Pull request by Casper3667**
Ports the ring bell from bay: https://github.com/Baystation12/Baystation12/pull/19995

### CODE REVIEW

**Comment by Casper3667** [comment]
!review

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -1037,6 +1038,22 @@
 	drop_sound = 'sound/items/drop/cardboardbox.ogg'
 	pickup_sound = 'sound/items/pickup/cardboardbox.ogg'
 
+/obj/item/toy/ringbell
+	name = "ringside bell"
+	desc = "A bell used to signal the beginning and end of various ring sports."
+	icon_state= "ringbell"
```
Comment (Geevies) [inline code/game/objects/items/toys.dm:1044]:
```suggestion
	icon_state = "ringbell"
```

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -1037,6 +1038,22 @@
 	drop_sound = 'sound/items/drop/cardboardbox.ogg'
 	pickup_sound = 'sound/items/pickup/cardboardbox.ogg'
 
+/obj/item/toy/ringbell
+	name = "ringside bell"
+	desc = "A bell used to signal the beginning and end of various ring sports."
```
Comment (Geevies) [inline code/game/objects/items/toys.dm:1043]: i would put a desc_info here explaining what the different intents signal

File: code/game/objects/items/toys.dm [hunk 3]
```diff
@@ -1037,6 +1038,22 @@
...
+	icon_state= "ringbell"
+	anchored = 1
```
Comment (Geevies) [inline code/game/objects/items/toys.dm:1045]:
```suggestion
	anchored = TRUE
```

File: code/game/objects/items/toys.dm [hunk 4]
```diff
@@ -1037,6 +1038,22 @@
...
+
+/obj/item/toy/ringbell/attack_hand(mob/user as mob)
```
Comment (Geevies) [inline code/game/objects/items/toys.dm:1047]:
```suggestion
/obj/item/toy/ringbell/attack_hand(mob/user)
```

File: code/game/objects/items/toys.dm [hunk 5]
```diff
@@ -1037,6 +1038,22 @@
...
+	if (user.a_intent == I_HELP)
```
Comment (Geevies) [inline code/game/objects/items/toys.dm:1048]: this can be made into a switch statement

File: code/game/objects/items/toys.dm [hunk 6]
```diff
@@ -1037,6 +1038,22 @@
...
+		user.visible_message("<span class='notice'>[user] rings \the [src], signalling the beginning of the contest.</span>")
```
Comment (Geevies) [inline code/game/objects/items/toys.dm:1049]: honestly, add FONT_LARGE to this too, and convert the straight HTML code to SPAN_NOTICE, same for the others

File: code/game/objects/items/toys.dm [hunk 7]
```diff
@@ -1037,6 +1038,22 @@
...
+		playsound(user.loc, 'sound/items/oneding.ogg', 60, 1)
+	else if (user.a_intent == I_DISARM)
+		user.visible_message("<span class='notice'>[user] rings \the [src] three times, signalling the end of the contest!</span>")
+		playsound(user.loc, 'sound/items/threedings.ogg', 60, 1)
+	else if (user.a_intent == I_HURT)
+		user.visible_message("<span class='warning'>[user] rings \the [src] repeatedly, signalling a disqualification!</span>")
```
Comment (Geevies) [inline code/game/objects/items/toys.dm:1055]: SPAN_WARNING

**Review by Geevies — CHANGES_REQUESTED** [review]

File: code/game/objects/items/toys.dm [hunk 8]
```diff
@@ -1041,19 +1041,21 @@
...
-		user.visible_message("<span class='warning'>[user] rings \the [src] repeatedly, signalling a disqualification!</span>")
-		playsound(user.loc, 'sound/items/manydings.ogg', 60, 1)
+	desc_info = "Use help intent on the bell to signal the start of a contest\ndisarm intent to signal the end of a contest and\nharm intent to signal a disqualification."
+	icon_state = "ringbell"
+	anchored = TRUE
+
+/obj/item/toy/ringbell/attack_hand(mob/user)
+	switch(user.a_intent)
+		if (I_HELP)
+			user.visible_message(FONT_LARGE (SPAN_NOTICE("[user] rings \the [src], signalling the beginning of the contest.")))
```
Comment (Geevies) [inline code/game/objects/items/toys.dm:1051]:
```suggestion
			user.visible_message(FONT_LARGE(SPAN_NOTICE("[user] rings \the [src], signalling the beginning of the contest.")))
```

File: code/game/objects/items/toys.dm [hunk 9]
```diff
@@ -1041,19 +1041,21 @@
...
+			playsound(user.loc, 'sound/items/oneding.ogg', 60, 1)
+		if (I_DISARM)
+			user.visible_message(FONT_LARGE (SPAN_NOTICE("[user] rings \the [src] three times, signalling the end of the contest!")))
```
Comment (Geevies) [inline code/game/objects/items/toys.dm:1054]:
```suggestion
			user.visible_message(FONT_LARGE(SPAN_NOTICE("[user] rings \the [src] three times, signalling the end of the contest!")))
```

File: code/game/objects/items/toys.dm [hunk 10]
```diff
@@ -1041,19 +1041,21 @@
...
+			playsound(user.loc, 'sound/items/threedings.ogg', 60, 1)
+		if (I_HURT)
+			user.visible_message(FONT_LARGE (SPAN_WARNING("[user] rings \the [src] repeatedly, signalling a disqualification!")))
```
Comment (Geevies) [inline code/game/objects/items/toys.dm:1057]:
```suggestion
			user.visible_message(FONT_LARGE(SPAN_WARNING("[user] rings \the [src] repeatedly, signalling a disqualification!")))
```

File: code/game/objects/items/toys.dm (hunk 8 above)
Comment (Geevies) [inline code/game/objects/items/toys.dm:1051]: should also add a line after this like "You ring \the [src], blahblahblah"

**Review by Geevies — CHANGES_REQUESTED** [review]

**Review by Geevies — APPROVED** [review]

**Review by fernerr — APPROVED** [review]

**Pull request merged**